### ISSUE TITLE
Continuing training from a partially trained model sometimes reverts learning-rate to previous value

### ISSUE THREAD

**Issue by tuglat** [awslabs/sockeye#890]
Experienced in both 2.1.16 and 2.1.21.

Training a model using --max-checkpoints to periodically stop and then continue training. When continuing training, the learn-rate seems to have been set back to the original. I have included the log files for two sequential training runs that exhibit this behavior.

This does not always happen, but I don't know how to characterize the different cases.

[log.00.txt](https://github.com/awslabs/sockeye/files/5329983/log.00.txt)
[log.01.txt](https://github.com/awslabs/sockeye/files/5329984/log.01.txt)

**Comment by tdomhan** [comment]
Thanks for reporting! Could you also share the full commands used for initially starting training and then to continue training?

**Comment by tuglat** [comment]
Whenever doing continued training w/ Sockeye 2 (last tested on 2.8.2), the LearningRateSchedulerPlateauReduce that I specify does not seem to be recovered from the saved state. Instead it appears that a new one is created. The learning rate is then set to the MXNet optimizer base learning rate, which **is** recovered, and set to the initial learning rate specified in the parameters.

Could this be related to the message 

      learning rate from ``lr_scheduler`` has been overwritten by ``learning_rate`` in optimizer.

that appears at the bottom of the the stdout from running sockeye-train? This message comes from mxnet/optimizer/optimizer.py.

**Comment by huache** [comment]
I encounter the same problem. Every time I continue the training, the learning rate will restart from 2e-4 ，and it will drastically reduce the performance of the model. I update the sockeye version from 2.1.13 to 2.1.16 , then 2.1.26 .  The problem still exists.

![image](https://user-images.githubusercontent.com/11642845/103450805-506e3e00-4cf6-11eb-8cd8-5a46fd09636f.png)

**Comment by tuglat** [comment]
This has to do w/ the fact that the optimizer is not saved in _save_trainer_states in training.py. The lr_scheduler is held inside the optimizer.

I am working on a fix that saves the lr_scheduler separately.

I closed the issue because it was marked as a question, and getting no attention. I was going to open it up a new issue, and submit my fix as a PR. Got sidetracked by family.

Steve

**Comment by huache** [comment]
Thank you so much !  Looking forward to your fix PR .

**Comment by tuglat** [comment]
@huache PR has been accepted and merged.

**Comment by tuglat** [comment]
fixed in PR's #920, #924, and #926

**Comment by HossamAmer12** [comment]
If I have a checkpoint (*.pt) model, can I continue training using sockeye? If yes, what's the command?
